kevm-pyk/src/kevm_pyk/foundry.py

```diff
@@ -421,7 +421,8 @@ def foundry_prove(
     break_on_jumpi: bool = False,
     break_on_calls: bool = True,
     implication_every_block: bool = True,
-    bmc_depth: int | None = None,
+    bmc_depth: int = 3,
+    unbounded_bmc_depth: bool = False,
```

Comment:
I find this a bit less intuitive than the original interface with `int | None`. (`unbounded_bmc_depth` is a self-contradiction.)

Reply:
> Would it be better if this other flag (e.g., `unbounded_loops`) just set the value of `bmc_depth` to `None`?

Yes, I think so. Luckily, in this particular case the, the CLI can be modified without changing the internal implementation.

Reply:
TY! Done in https://github.com/runtimeverification/evm-semantics/pull/1963/commits/38239deaec847e0ee10bdcce31d04297c649ae4c.
Now `unbounded_loops` is referenced once in `exec_foundry_prove`, where it's used to set `bmc_depth` to `None`—hope this looks better.